src/elementary/SpecialDataID/src/ctor.hpp

```diff
@@ -0,0 +1,18 @@
+/**
+ *  @brief Constructor
+ *
+ *  @param[in] incident    the incident particle
+ *  @param[in] target      the target
+ *  @param[in] type        the special data type
+ */
+SpecialDataID( const ParticleID& incident, const ParticleID& target,
+               const SpecialDataType& type ) :
+  Identifier( ParticlePairID( incident, target ).symbol() + "->" + type.name(),
+              false ) {}
+
+/**
+ *  @brief Constructor
+ *
+ *  @param[in] string   the identifier string
+ */
+SpecialDataID( const std::string& string ) : Identifier( string ) {}
```

Comment:
You have a validate function, but it's not called in the constructor.  Is it called some other way that I'm missing?

Reply:
It is used in the Identifier constructor.